src/main/java/by/antonsh/tictactoe/Tictactoe.java

```diff
@@ -0,0 +1,170 @@
+
+                                }else if(res==FILED_X*3){
+                                    JOptionPane.showMessageDialog(this,"Крестики WIN","Победа",JOptionPane.INFORMATION_MESSAGE);
+                                    System.exit(0);
+                                }else{
+                                    JOptionPane.showMessageDialog(this,"Ничья","Ничья",JOptionPane.INFORMATION_MESSAGE);
+                                    System.exit(0);
+                                }
+                            }
+                        }
+                    }
+
+                }
+
+                @Override
+                protected void paintComponent(Graphics g) {
+                    super.paintComponent(g);
+                    drawGrid(g);
+                    drawXO(g);
+
+
+                }
+                void drawGrid(Graphics g){
+                    int w=getWidth();
+                    int h=getHeight();
+                    int dw=w/3;
+                    int dh=h/3;
+                    g.setColor(Color.BLUE);
+                    for (int i = 0; i < 3; i++) {
+                        g.drawLine(0,dh*i,w,dh*i);
+                        g.drawLine(dw*i,0,dw*i,h);
+                    }
+                }
+                void drawX(int k,int m,Graphics g){
+
+                    int dw=getWidth()/3;
+                    int dh=getHeight()/3;
+                    g.setColor(Color.BLACK);
+                    int x=k*dw;
+                    int y=m*dh;
+                    g.drawLine(x,y,x+dw,y+dh);
+                    g.drawLine(x,y+dh,x+dw,y);
+                }
+                void drawO(int k,int m,Graphics g){
+
+                    int dw=getWidth()/3;
+                    int dh=getHeight()/3;
+                    g.setColor(Color.BLACK);
+                    int x=k*dw;
+                    int y=m*dh;
+                    g.drawOval(x+5*dw/100,y,dw*9/10,dh);
+                }
+                void drawXO(Graphics g){
+                    for (int i = 0; i < 3; ++i) {
+                        for (int j = 0; j < 3; ++j) {
+                            if(filed[i][j]==FILED_X){
+                                drawX(i,j,g);
+                            }else if (filed[i][j]==FILED_0){
+                                drawO(i,j,g);
+                            }
+                        }
+                    }
+                }
+            }
+            @Override
+            public void run() {
+                JFrame window=new JFrame("Tictactoe");
+                window.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+                window.setSize(450,450);
+                window.setLayout(new BorderLayout());
+                window.setLocationRelativeTo(null);
+                window.setVisible(true);
+                Tic game=new Tic();
+                window.add(game);
+            }
+            
+        });
+        th1.start();
```

Comment:
this is a very difficult decision. if you can make something simpler, you need to do it, you should not complicate it and add unnecessary functions. the code should be as clear as possible